src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/germline/IntegerCopyNumberExpectationsCalculator.java

```diff
@@ -0,0 +1,342 @@
+     */
+    private final static boolean CHECK_FOR_NANS = false;
+
+    /**
+     * If true, CN = 0 state will be excluded in calculating the posteriors. The copy ratio posterior
+     * probabilities will be renormalized to unity.
+     */
+    private final static boolean NEGLECT_ZERO_COPY_NUMBER_STATE = true;
+
+    private final Set<String> sexGenotypesSet;
+
+    /**
+     * A map from each sex genotype to a corresponding HMM
+     */
+    private final Map<String, IntegerCopyNumberHiddenMarkovModel<CoverageModelCopyRatioEmissionData>> hmm;
+
+    /**
+     * Integer copy number transition probability and prior cache
+     */
+    private final IntegerCopyNumberTransitionProbabilityCacheCollection cache;
+
+    public IntegerCopyNumberExpectationsCalculator(
+            @Nonnull final IntegerCopyNumberTransitionProbabilityCacheCollection cache,
+            final int readCountThresholdPoissonSwitch) {
+        this.cache = Utils.nonNull(cache, "The integer copy number transition probability cache" +
+                " collection must be non-null");
+        /* create an integer copy number HMM for each sex genotype in the collection */
+        sexGenotypesSet = cache.getSexGenotypes();
+        hmm = new HashMap<>();
+        final CoverageModelCopyRatioEmissionProbabilityCalculator emissionProbabilityCalculator =
+                new CoverageModelCopyRatioEmissionProbabilityCalculator(readCountThresholdPoissonSwitch);
+        sexGenotypesSet.forEach(sexGenotype -> hmm.put(sexGenotype,
+                new IntegerCopyNumberHiddenMarkovModel<>(emissionProbabilityCalculator, cache, sexGenotype)));
+    }
+
+    @Override
+    public CopyRatioExpectations getCopyRatioPosteriorExpectations(
+            @Nonnull final CopyRatioCallingMetadata copyRatioCallingMetadata,
+            @Nonnull final List<Target> targetList,
+            @Nonnull final List<CoverageModelCopyRatioEmissionData> emissionData) {
+        verifyArgsForPosteriorExpectations(copyRatioCallingMetadata, targetList, emissionData);
+
+        /* choose the HMM */
+        final IntegerCopyNumberHiddenMarkovModel<CoverageModelCopyRatioEmissionData> genotypeSpecificHMM =
+                hmm.get(copyRatioCallingMetadata.getSampleSexGenotypeData().getSexGenotype());
+
+        /* supplement the emission data with sample metadata */
+        emissionData.forEach(datum -> datum.setCopyRatioCallingMetadata(copyRatioCallingMetadata));
+
+        /* run the forward-backward algorithm */
+        final ForwardBackwardAlgorithm.Result<CoverageModelCopyRatioEmissionData, Target, IntegerCopyNumberState> result =
+                ForwardBackwardAlgorithm.apply(emissionData, targetList, genotypeSpecificHMM);
+
+        /* indices of hidden states to include in calculating posterior expectations */
+        final List<IntegerCopyNumberState> hiddenStates = genotypeSpecificHMM.hiddenStates();
+        final int[] includedHiddenStatesIndices;
```

Comment:
I got rid of `NEGLECT_ZERO_COPY_NUMBER_STATE` altogether in the post-PR commit. Good suggestion though!